packages/ui/nodes/image/src/ImageElement/ImageElement.tsx

```diff
@@ -1,210 +1,94 @@
-            maxWidth="100%"
-            lockAspectRatio
-            resizeRatio={align === 'center' ? 2 : 1}
-            enable={{
-              left: ['center', 'left'].includes(align),
-              right: ['center', 'right'].includes(align),
-            }}
             handleComponent={{
               left: (
-                <ImageHandle
+                <Box
                   css={[styles.handleLeft?.css]}
                   className={styles.handleLeft?.className}
                 />
               ),
               right: (
-                <ImageHandle
+                <Box
                   css={styles.handleRight?.css}
                   className={styles.handleRight?.className}
                 />
               ),
             }}
-            handleStyles={{
-              left: { left: 0 },
-              right: { right: 0 },
-            }}
-            onResize={(e, direction, ref) => {
-              setWidth(ref.offsetWidth);
-            }}
-            onResizeStop={(e, direction, ref) => setNodeWidth(ref.offsetWidth)}
-            {...resizeProps}
+            ignoreReadOnly={ignoreReadOnly}
+            align={align}
+            {...resizableProps}
           >
-            <img
+            <ImageImg
               data-testid="ImageElementImage"
               css={styles.img?.css}
               className={styles.img?.className}
-              src={url}
-              alt={captionString}
-              draggable={draggable}
               {...nodeProps}
             />
-          </Resizable>
+          </ImageResizable>
 
-          {!caption.disabled && (captionString.length || selected) && (
-            <figcaption
-              data-testid="ImageElementCaption"
-              style={{ width }}
-              css={styles.figcaption?.css}
-              className={styles.figcaption?.className}
-            >
-              <TextareaAutosize
-                data-testid="ImageElementTextArea"
-                css={styles.caption?.css}
-                className={styles.caption?.className}
-                value={(nodeCaption[0] as TText).text}
-                placeholder={placeholder}
-                onChange={onChangeCaption}
-                readOnly={(!ignoreReadOnly && readOnly) || caption.readOnly}
-              />
-
-              {/* <div css={styles.caption?.css}> */}
-              {/*  <Plate */}
-              {/*    id={captionId} */}
-              {/*    plugins={plugins} */}
-              {/*    initialValue={nodeCaption} */}
-              {/*    value={nodeCaption} */}
-              {/*    editableProps={{ */}
-              {/*      placeholder, */}
-              {/*      className: styles.caption?.className, */}
-              {/*    }} */}
-              {/*    onChange={onChangeCaption} */}
-              {/*  /> */}
-              {/* </div> */}
-            </figcaption>
-          )}
+          <ImageCaption
+            data-testid="ImageElementCaption"
```

Comment:
What your thoughts on more semantic data attr?

Perhaps something like `data-comp-id`.

My personal preference is to avoid testing specific logic in the code if possible.

Reply:
How testing-react can query it without data-testid?